pkg/virtctl/spice/spice.go

```diff
@@ -0,0 +1,105 @@
+package spice
+
+import (
+	"errors"
+	"fmt"
+	flag "github.com/spf13/pflag"
+	"io/ioutil"
+	kubev1 "k8s.io/client-go/pkg/api/v1"
+	"k8s.io/client-go/rest"
+	"kubevirt.io/kubevirt/pkg/kubecli"
+	"log"
+	"os"
+	"os/exec"
+)
+
+const FLAG = "spice"
+const TEMP_PREFIX = "spice"
+
+type Spice struct {
+}
+
+func DownloadSpice(vm string, restClient *rest.RESTClient) (string, error) {
+	body, err := restClient.Get().
+		Resource("vms").SetHeader("Accept", "text/plain").
+		SubResource("spice").
+		Namespace(kubev1.NamespaceDefault).
+		Name(vm).Do().Raw()
+	if err != nil {
+		return "", errors.New(fmt.Sprintf("Can't read body: %s\n", err.Error()))
+	}
+	return fmt.Sprintf("%s", body), nil
+}
+
+func (o *Spice) FlagSet() *flag.FlagSet {
+
+	cf := flag.NewFlagSet(FLAG, flag.ExitOnError)
+	cf.BoolP("details", "d", false, "If present, print SPICE console to stdout, otherwise run remote-viewer")
+	return cf
+}
+
+func (o *Spice) Run(flags *flag.FlagSet) int {
+	server, _ := flags.GetString("server")
+	kubeconfig, _ := flags.GetString("kubeconfig")
+	details, _ := flags.GetBool("details")
+
+	if len(flags.Args()) != 2 {
+		log.Println("VM name is missing")
+		return 1
+	}
+	vm := flags.Arg(1)
+
+	restClient, err := kubecli.GetRESTClientFromFlags(server, kubeconfig)
+
+	if err != nil {
+		log.Println(err)
+		return 1
+	}
+	body, err := DownloadSpice(vm, restClient)
+	if err != nil {
+		log.Fatalf(err.Error())
+		return 1
+	}
+	if details {
+		fmt.Printf("%s", body)
+	} else {
+		f, err := ioutil.TempFile("", TEMP_PREFIX)
+
+		if err != nil {
+			log.Fatalf("Can't open file: %s", err.Error())
+			return 1
+		}
+		defer f.Close()
+
+		_, err = f.WriteString(body)
+		if err != nil {
+			log.Fatalf("Can't write to file: %s", err.Error())
+			defer os.Remove(f.Name())
+			return 1
+		}
+
+		f.Sync()
+
+		cmnd := exec.Command("remote-viewer", f.Name())
+		err = cmnd.Run()
+		defer os.Remove(f.Name())
```

Comment:
This defer is too late, it needs to be directly after `defer f.Close()`, the rest looks good to me :)